File: container_mgmt/lib/models/bay.dart
class Bay {
  final int bayId;
  final String bayNumber;
  final int blockId;

  Bay({required this.bayId, required this.bayNumber, required this.blockId});

  factory Bay.fromJson(Map<String, dynamic> json) => Bay(
    bayId: json['bayId'],
    bayNumber: json['bayNumber'],
    blockId: json['blockId'],
  );
}

File: con_mgmt_api/Models/Bay.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ContainerManagementApi.Models;

[Table("Bays")]
public class Bay
{
    [Key]
    public int BayId { get; set; }

    [Required]
    [MaxLength(10)]
    public string BayNumber { get; set; } = string.Empty;

    [Required]
    public int BlockId { get; set; }

    // Navigation properties removed to prevent circular references
}
